test/framework/machinedeployment_helpers.go

```diff
@@ -530,7 +530,11 @@ func ScaleAndWaitMachineDeploymentTopology(ctx context.Context, input ScaleAndWa
 	Expect(input.Cluster.Spec.Topology.Workers.MachineDeployments).NotTo(BeEmpty(), "Invalid argument. input.Cluster must have at least one MachineDeployment topology")
 
 	mdTopology := input.Cluster.Spec.Topology.Workers.MachineDeployments[0]
-	log.Logf("Scaling machine deployment topology %s from %d to %d replicas", mdTopology.Name, *mdTopology.Replicas, input.Replicas)
+	if mdTopology.Replicas != nil {
```

Comment:
The ScaleAndWaitMachineDeploymentTopology is being called with replicas. In this change we are only doing the nil check on the "before replicas" so that we can have a proper log statement.